guillotina/contrib/memcached/driver.py

```diff
@@ -92,25 +119,29 @@ async def set(self, key: str, data: str, *, expire: Optional[int] = None) -> Non
         kwargs: Dict[Any] = {}
         if expire is not None:
             kwargs["exptime"] = expire
-        await client.set(key.encode(), data.encode(), **kwargs)
+        with watch("set"):
+            await client.set(key.encode(), data.encode(), **kwargs)
 
     async def get(self, key: str) -> Optional[bytes]:
         client = self._get_client()
-        item: Optional[emcache.Item] = await client.get(key.encode())
+        with watch("get"):
+            item: Optional[emcache.Item] = await client.get(key.encode())
         if item is not None:
             return item.value
         else:
             return None
 
     async def delete(self, key: str) -> None:
         client = self._get_client()
-        await client.delete(key.encode())
+        with watch("delete"):
+            await client.delete(key.encode())
 
     async def delete_all(self, keys: List[str]) -> None:
         client = self._get_client()
         for key in keys:
             try:
-                await client.delete(key.encode())
+                with watch("delete_many"):
```

Comment:
These are all single deletes!
```suggestion
                with watch("delete"):
```

Reply:
Maybe you could has a watch outside the for loop to measure how long deleting all the keys inside a request take?

Reply:
good catch! yes